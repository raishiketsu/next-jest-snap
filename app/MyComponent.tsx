"use client"
const MyComponent: React.FC = () => {
  return (
    <div>
      <h3>Hello, Jest!</h3>
    </div>
  )
};

export default MyComponent;