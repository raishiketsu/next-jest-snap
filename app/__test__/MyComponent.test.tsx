import { render, screen } from '@testing-library/react';
import MyComponent from '../MyComponent';

describe('MyComponent', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<MyComponent />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(<MyComponent />);
    const linkElement = screen.getByText(/Hello, Jest!/i);
    expect(linkElement).toBeInTheDocument();
  });
});
