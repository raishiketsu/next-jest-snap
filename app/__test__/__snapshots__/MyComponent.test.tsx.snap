// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Hello, Jest!
    </h3>
  </div>
</DocumentFragment>
`;

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Hello, Jest!
    </h3>
  </div>
</DocumentFragment>
`;
